stub_uploader/metadata.py

```diff
@@ -174,6 +177,43 @@ def verify_typeshed_req(req: Requirement) -> None:
 }
 
 
+def validate_response(resp: requests.Response, req: Requirement) -> None:
+    if resp.status_code != 200:
+        raise InvalidRequires(
+            f"Expected dependency {req} to be accessible on PyPI, but got {resp.status_code}"
+        )
+
+
+def get_sdist_requires(
+    sdist_data: dict[str, str], req: Requirement
+) -> Generator[Requirement, None, None]:
+    filename = sdist_data["filename"]
+    resp = requests.get(sdist_data["url"], stream=True)
+    validate_response(resp, req)
+    with open(filename, "wb") as file:
+        file.write(resp.raw.read())
+
+    folder_name = ""
+    with tarfile.open(filename) as file_in:
+        file_in.extractall()
+        for info in file_in:
+            folder_name = info.name
+            # Assume a single folder
+            break
```

Comment:
I would add a check here, whether `folder_name` was actually set, or maybe even check that there really is exactly one folder to account for broken eggs. But for the former case, the following should be simpler and have that required error check:

```suggestion
        # Assume a single folder
        folder_name = next(iter(file_in)).name
```

You can then also drop the initialization of `folder_name`.

Or just use `*` instead of `folder_name` in the glob below.